File: src/ByteBite.Infrastructure/Persistence/Entities/Product.cs
using System;
using System.Collections.Generic;

namespace ByteBite.Infrastructure.Persistence.Entities;

/// <summary>
/// 商品表
/// </summary>
public partial class Product
{
    public Guid Id { get; set; }

    public Guid StoreId { get; set; }

    public Guid CategoryId { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    /// <summary>
    /// 基础价格（默认规格价格）
    /// </summary>
    public decimal BasePrice { get; set; }

    public string? ImageUrl { get; set; }

    /// <summary>
    /// 状态：on-上架, off-下架, sold_out-售罄
    /// </summary>
    public string Status { get; set; } = null!;

    public int SortOrder { get; set; }

    /// <summary>
    /// 最低起购数量
    /// </summary>
    public int MinOrderQty { get; set; }

    /// <summary>
    /// 月销量（冗余，定时统计）
    /// </summary>
    public int MonthlySales { get; set; }

    /// <summary>
    /// 总销量
    /// </summary>
    public int TotalSales { get; set; }

    /// <summary>
    /// 是否为套餐商品
    /// </summary>
    public bool IsCombo { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual Category Category { get; set; } = null!;

    public virtual ICollection<ComboItem> ComboItemComboProducts { get; set; } = new List<ComboItem>();

    public virtual ICollection<ComboItem> ComboItemProducts { get; set; } = new List<ComboItem>();

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();

    public virtual ICollection<ProductSalesStat> ProductSalesStats { get; set; } = new List<ProductSalesStat>();

    public virtual ICollection<SpecGroup> SpecGroups { get; set; } = new List<SpecGroup>();

    public virtual Store Store { get; set; } = null!;
}

File: web/src/types/models/product.ts
/** 规格项DTO */
export interface SpecOptionDto {
  /** 规格项ID */
  id: string
  /** 规格项名称 */
  name: string
  /** 加价 */
  extraPrice: number
  /** 库存（null表示不限库存） */
  stock?: number
  /** 是否为默认选中项 */
  isDefault: boolean
}

/** 规格组DTO */
export interface SpecGroupDto {
  /** 规格组ID */
  id: string
  /** 规格组名称 */
  name: string
  /** 是否必选 */
  isRequired: boolean
  /** 规格项列表 */
  options: SpecOptionDto[]
}

/** 商品信息DTO */
export interface ProductDto {
  /** 商品ID */
  id: string
  /** 所属店铺ID */
  storeId: string
  /** 所属分类ID */
  categoryId: string
  /** 分类名称 */
  categoryName: string
  /** 商品名称 */
  name: string
  /** 商品描述 */
  description?: string
  /** 基础价格 */
  basePrice: number
  /** 商品图片URL */
  imageUrl?: string
  /** 状态：on-上架, off-下架, sold_out-售罄 */
  status: string
  /** 排序序号 */
  sortOrder: number
  /** 最低起购数量 */
  minOrderQty: number
  /** 月销量 */
  monthlySales: number
  /** 总销量 */
  totalSales: number
  /** 是否为套餐商品 */
  isCombo: boolean
  /** 规格组列表 */
  specGroups: SpecGroupDto[]
}

/** 创建规格项请求 */
export interface CreateSpecOptionRequest {
  /** 规格项名称 */
  name: string
  /** 加价 */
  extraPrice: number
  /** 库存（null表示不限库存） */
  stock?: number
  /** 是否为默认选中项 */
  isDefault: boolean
}

/** 创建规格组请求 */
export interface CreateSpecGroupRequest {
  /** 规格组名称 */
  name: string
  /** 是否必选 */
  isRequired?: boolean
  /** 规格项列表 */
  options: CreateSpecOptionRequest[]
}

/** 创建商品请求 */
export interface CreateProductRequest {
  /** 所属分类ID */
  categoryId: string
  /** 商品名称 */
  name: string
  /** 商品描述 */
  description?: string
  /** 基础价格 */
  basePrice: number
  /** 商品图片URL */
  imageUrl?: string
  /** 最低起购数量 */
  minOrderQty?: number
  /** 是否为套餐商品 */
  isCombo?: boolean
  /** 规格组列表 */
  specGroups: CreateSpecGroupRequest[]
}

/** 更新商品请求 */
export interface UpdateProductRequest {
  /** 所属分类ID */
  categoryId?: string
  /** 商品名称 */
  name?: string
  /** 商品描述 */
  description?: string
  /** 基础价格 */
  basePrice?: number
  /** 商品图片URL */
  imageUrl?: string
  /** 状态：on-上架, off-下架, sold_out-售罄 */
  status?: string
  /** 排序序号 */
  sortOrder?: number
  /** 最低起购数量 */
  minOrderQty?: number
}
